src/main/java/org/killbill/billing/plugin/core/resources/PluginHealthcheck.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2014-2017 Groupon, Inc
+ * Copyright 2014-2017 The Billing Project, LLC
+ *
+ * The Billing Project licenses this file to you under the Apache License, version 2.0
+ * (the "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at:
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.killbill.billing.plugin.core.resources;
+
+import java.util.Dictionary;
+
+import javax.annotation.Nullable;
+
+import org.jooby.MediaType;
+import org.jooby.Result;
+import org.jooby.Results;
+import org.jooby.Status;
+import org.killbill.billing.plugin.service.Healthcheck;
+import org.killbill.billing.plugin.service.Healthcheck.HealthStatus;
+import org.killbill.billing.tenant.api.Tenant;
+
+// Specified in the concrete class
+//@Singleton
+//@Path("/healthcheck")
+public abstract class PluginHealthcheck {
```

Comment:
Couple of questions:
1. Why is this class `abstract` -- would it make sense to then have some abstract methods ?
2. Do we expect KB core to call this plugin healthcheck endpoints to aggregate a global result (core + plugins), or do we expect the client  (LB, ...) to call multiple endpoints?
3. Would it make sense to instead have an OSGI healtcheck service and keep one endpoint at the KB core level?